polyfill/lib/ecmascript.mjs

```diff
@@ -2837,15 +2841,21 @@ export const ES = ObjectAssign({}, ES2020, {
         // no-op
         break;
       case 'months':
-        if (!calendar) throw new RangeError('a starting point is required for months balancing');
-        // balance years down to months
-        while (MathAbs(years) > 0) {
-          const newRelativeTo = ES.CalendarDateAdd(calendar, relativeTo, oneYear, {});
-          const oneYearMonths = ES.CalendarDateUntil(calendar, relativeTo, newRelativeTo, { largestUnit: 'months' })
-            .months;
-          relativeTo = newRelativeTo;
-          months += oneYearMonths;
-          years -= sign;
+        {
+          if (!calendar) throw new RangeError('a starting point is required for months balancing');
+          // balance years down to months
+          const dateAdd = ES.GetMethod(calendar, 'dateAdd');
```

Comment:
This and all following similar modifications change two things semantically:
- No more `if (!ES.IsTemporalDate(result)) throw new TypeError('invalid result');` aka `Perform ? RequireInternalSlot(addedDate, [[InitializedTemporalDate]]).` (not good?)
- Reduced lookups (good)

I think the spec doesn't match either (same with second instance in later comment, from previous optimization)